src/backend/web/routes/posts.js

```diff
@@ -98,7 +98,11 @@ posts.get('/:id', async (req, res) => {
           res.append('Content-type', 'text/html').send(post.html);
           break;
         default:
-          res.append('Content-type', 'application/json').json(post);
+          if (accept.type() !== 'json' || accept.type() !== 'text' || accept.type() !== 'html') {
```

Comment:
You can remove this `if`, and always send your 406, since you've checked that it's not these cases above in your switch cases (i.e., if it gets to this `default` section, it's none of the ones you want).

Reply:
@humphd I was trying different stuff and forgot to remove this if statement. I'll commit the code in a few minutes.
 
Here are my thoughts about tests for this issue: 

So I could write 3 test cases to check every valid content type. I already have a basic understanding how jest tests should look like. I know that I need to use `expect(res.status).toEqual(200)` if I'll write those 3 tests. But I also need to create 3 posts with different data types - JSON, plain text and HTML. JSON is easy, I saw it in the code already so I know how to create and test it. But I don't understand how to create HTML and plain text posts. 

Or I could just write one test that will expect a 406 error and check that content type is not a valid one. For this case I also need to create a post that is not JSON, html and text. What type should I use then? How can I create it?

Reply:
You should write 4 tests, similar to what I did manually below:

1. send request with `Accept: text/plain` and expect to get back a 200 response with `Content-Type: text/plain`
1. send request with `Accept: text/html` and expect to get back a 200 response with `Content-Type` that starts with `text/html`
1. send request with `Accept: application/json` and expect to get back a 200 response with `Content-Type: application/json`

For the 4th test, you need to do:

send a request with an invalid type, for example: `Accept: image/jpg` and expect to get back a 406 response.

In terms of how to do this kind of test, you can do something like this:

```js
  it('responds with json when requesting json data', () => { 
    const res = await request(app).get(`/posts/${hash(existingGuid)}`).set('Accept', 'application/json');

      expect(res.get('Content-Type')).toContain('application/json')
      expect(res.status).toBe(200);
});
```

See docs in https://www.npmjs.com/package/supertest#example and an existing test that is similar in https://github.com/Seneca-CDOT/telescope/blob/master/test/posts.test.js#L106-L113